lib/routes/utils/request_helper.js

```diff
@@ -80,8 +82,61 @@ function shouldSendVerifyAccountEmail(account, request) {
   return (!context || !!sendEmailIfUnverified) && !account.emailVerified
 }
 
+/**
+ * Returns whether or not to use signin unblock feature on a request.
+ *
+ * @param account
+ * @param config
+ * @param request
+ * @returns {boolean}
+ */
+function shouldEnableSigninUnblock(account, config, request) {
+
+  var enabled = config.signinUnblock && config.signinUnblock.enabled
+  if (!enabled) {
+    return false
+  }
+
+  //if (request.app.isSuspiciousRequest) {
+  //  return true
+  //}
+
+  // Or if the email address matching one of these regexes.
+  var email = account.email
+  var isValidEmail = config.signinUnblock.allowEmailRegex.some(function (reg) {
+    var emailReg = new RegExp(reg)
+    return emailReg.test(email)
+  })
+
+  if (isValidEmail) {
+    return true
+  }
+
+  // While we're testing this feature, there may be some funky
+  // edge-cases in device login flows that haven't been fully tested.
+  // Temporarily avoid them for regular users by checking the `context` flag,
+  // This check may go away in the final version of this feature.
```

Comment:
IIUC the comment here is inaccurate for this feature, in that it's not likely to go away in the final version - there will always be clients that can't support this unblock flow.